Travis: update to focal dist

0xc0170 opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

Updating to the latest Travis provided environment. It's been a year around and we can benefit from
more updated versions we use.

More info https://docs.travis-ci.com/user/reference/focal/

The default one for Travis is xenial. The one we also have been using for 2 years now. Should we update to focal? To get updates to the tools. If tests pass, anything against this update?

As we update, I need to fix outstanding issues in Travis - littlefs should not be part of our testing here, plus any 20.04 issues with packages.

What this brings:
- update to 20.04 dist in Travis
- removing old tools check with python 2.7, tools are frozen anyway
- removing littlefs generic tests (we have integrations tests, and the library itself it's tested by their own tests)

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

0xc0170 committed:
Travis: update to 20.04

Updating to the latest Travis provided environment. It's been a year around and we can benefit from
more updated versions we use.

More info https://docs.travis-ci.com/user/reference/focal/

Patater commented on .travis.yml:
```diff
@@ -16,7 +16,7 @@
 language: sh
 os: linux
-dist: xenial
+dist: focal
```
PR https://github.com/ARMmbed/mbed-os/pull/14538 adds CMake based on the Ubuntu version, so this would break it. Just a heads up we'll need to coordinate merging those.

0xc0170 committed:
Travis: remove littlefs generic tests

They do not add any value as littlefs library contains own tests, we do not need to duplicate them in Mbed OS.
We test integration with our libraries.

0xc0170 committed:
aspell: fix dependencies missing

As of ubuntu 20.04, the two deps for aspell are missing.

0xc0170 committed:
fixup! aspell: fix dependencies missing

0xc0170 committed:
fixup! aspell: fix dependencies missing

0xc0170 committed:
Travis: deprecate python 2.7

Python 3, it's your time now

LDong-Arm commented on .travis.yml:
```diff
@@ -216,10 +217,10 @@ matrix:
-      name: "tools-py27"
-      env: NAME=tools-py2.7
+      name: "tools-py35"
+      env: NAME=tools-py3.5
       language: python
-      python: 2.7
+      python: 3.5
```
Python 3.5 has also reached end-of-life. Since it's a recent event (late 2020), it's okay to keep or remove it I guess.

rajkan01 reviewed (dismissed):
LGTM

LDong-Arm reviewed (approved):
LGTM, thanks for the clean up

LDong-Arm commented on .travis.yml:
```diff
@@ -138,6 +138,7 @@ matrix:
       name: "spellcheck"
       env: NAME=doxy-spellcheck
       install:
+        - sudo apt-get -y install libncursesw5
```
Not crucial, but we can always install apt packages in a more declarative way: https://github.com/ARMmbed/mbed-os-example-blinky/blob/0016df36542bdcb2c84e4902b9c50364da8120a8/.travis.yml#L33-L41

0xc0170 replied:
Added it via addons in the update

0xc0170 committed:
Travis: use addons to install libncursesw5
